File: data/compilers_comparation/src/examples/second_example/src/func34.c
#include "second_example.h" 
array_t* func34(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array64;
   if (pCounter > 0) {
      array64 = vars->data[--pCounter];
      array64->refC++;
      DEBUG_COPY(array64->id);
   } else {
      array64 = (array_t*)malloc(sizeof(array_t));
      array64->size = 590;
      array64->refC = 1;
      array64->id = 64;
      array64->data = (unsigned int*)malloc(array64->size*sizeof(unsigned int));
      memset(array64->data, 0, array64->size*sizeof(unsigned int));
      DEBUG_NEW(array64->id);
   }
   unsigned int loop21 = 0;
   unsigned int loopLimit21 = (rand()%loopsFactor)/4 + 1;
   for(; loop21 < loopLimit21; loop21++) {
      if(PATH0 & 1) {
         array_t* array65;
         if (pCounter > 0) {
            array65 = vars->data[--pCounter];
            array65->refC++;
            DEBUG_COPY(array65->id);
         } else {
            array65 = (array_t*)malloc(sizeof(array_t));
            array65->size = 505;
            array65->refC = 1;
            array65->id = 65;
            array65->data = (unsigned int*)malloc(array65->size*sizeof(unsigned int));
            memset(array65->data, 0, array65->size*sizeof(unsigned int));
            DEBUG_NEW(array65->id);
         }
         unsigned int loop22 = 0;
         unsigned int loopLimit22 = (rand()%loopsFactor)/5 + 1;
         for(; loop22 < loopLimit22; loop22++) {
            for (int i = 0; i < array65->size; i++) {
               array65->data[i]--;
            }
            for (int i = 0; i < array64->size; i++) {
               if (array64->data[i] == 86) { 
                  return array64;
               }
            }
         }
         array_t_param params0;
         params0.size = 2;
         params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
         params0.data[0] = array64;
         params0.data[1] = array65;
         array_t* array66 = func54(&params0, rng(), loopsFactor);
         DEBUG_RETURN(array66->id);
         free(params0.data);
         array66->refC--;
         if(array66->refC == 0) {
            free(array66->data);
            free(array66);
            DEBUG_FREE(array66->id);
         }
         array65->refC--;
         if(array65->refC == 0) {
            free(array65->data);
            free(array65);
            DEBUG_FREE(array65->id);
         }
      }
      else {
         array_t_param params0;
         params0.size = 1;
         params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
         params0.data[0] = array64;
         array_t* array67 = func50(&params0, loopsFactor);
         DEBUG_RETURN(array67->id);
         free(params0.data);
         array67->refC--;
         if(array67->refC == 0) {
            free(array67->data);
            free(array67);
            DEBUG_FREE(array67->id);
         }
      }
      array_t* array68;
      if (pCounter > 0) {
         array68 = vars->data[--pCounter];
         array68->refC++;
         DEBUG_COPY(array68->id);
      } else {
         array68 = (array_t*)malloc(sizeof(array_t));
         array68->size = 197;
         array68->refC = 1;
         array68->id = 68;
         array68->data = (unsigned int*)malloc(array68->size*sizeof(unsigned int));
         memset(array68->data, 0, array68->size*sizeof(unsigned int));
         DEBUG_NEW(array68->id);
      }
      array_t_param params0;
      params0.size = 2;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array64;
      params0.data[1] = array68;
      array_t* array69 = func51(&params0, loopsFactor);
      DEBUG_RETURN(array69->id);
      free(params0.data);
      if(PATH0 & 2) {
         array_t* array70;
         if (pCounter > 0) {
            array70 = vars->data[--pCounter];
            array70->refC++;
            DEBUG_COPY(array70->id);
         } else {
            array70 = (array_t*)malloc(sizeof(array_t));
            array70->size = 804;
            array70->refC = 1;
            array70->id = 70;
            array70->data = (unsigned int*)malloc(array70->size*sizeof(unsigned int));
            memset(array70->data, 0, array70->size*sizeof(unsigned int));
            DEBUG_NEW(array70->id);
         }
         unsigned int loop23 = 0;
         unsigned int loopLimit23 = (rand()%loopsFactor)/5 + 1;
         for(; loop23 < loopLimit23; loop23++) {
            for (int i = 0; i < array64->size; i++) {
               array64->data[i]--;
            }
            for (int i = 0; i < array70->size; i++) {
               if (array70->data[i] == 22) { 
                  return array70;
               }
            }
         }
         array70->refC--;
         if(array70->refC == 0) {
            free(array70->data);
            free(array70);
            DEBUG_FREE(array70->id);
         }
      }
      else {
         array_t_param params1;
         params1.size = 3;
         params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
         params1.data[0] = array64;
         params1.data[1] = array68;
         params1.data[2] = array69;
         array_t* array71 = func52(&params1, loopsFactor);
         DEBUG_RETURN(array71->id);
         free(params1.data);
         array71->refC--;
         if(array71->refC == 0) {
            free(array71->data);
            free(array71);
            DEBUG_FREE(array71->id);
         }
      }
      for (int i = 0; i < array64->size; i++) {
         if (array64->data[i] == 46) { 
            return array64;
         }
      }
      array69->refC--;
      if(array69->refC == 0) {
         free(array69->data);
         free(array69);
         DEBUG_FREE(array69->id);
      }
      array68->refC--;
      if(array68->refC == 0) {
         free(array68->data);
         free(array68);
         DEBUG_FREE(array68->id);
      }
   }
   for (int i = 0; i < array64->size; i++) {
      array64->data[i]--;
   }
   for (int i = 0; i < array64->size; i++) {
      if (array64->data[i] == 22) { 
         return array64;
      }
   }
   return array64;
}

File: data/multilang_comparison/program_nim/src/func34.nim
proc func34*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array94: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array94 = vars.data[pCounter]
      inc array94.refC
      debugCopy(array94.id)
   else:
      array94 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array94.size = 320
      array94.refC = 1
      array94.id = 94
      array94.data = cast[ptr UncheckedArray[uint32]](alloc(array94.size * sizeof(uint32)))
      zeroMem(array94.data, array94.size * sizeof(uint32))
      debugNew(array94.id)
   let loopLimit30 = (50) div (4) + 1
   for loop30 in 0 ..< loopLimit30:
      discard
      if (PATH0 and 1'u64) != 0:
         discard
         var array95: ptr ArrayT
         if pCounter > 0:
            dec pCounter
            array95 = vars.data[pCounter]
            inc array95.refC
            debugCopy(array95.id)
         else:
            array95 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
            array95.size = 736
            array95.refC = 1
            array95.id = 95
            array95.data = cast[ptr UncheckedArray[uint32]](alloc(array95.size * sizeof(uint32)))
            zeroMem(array95.data, array95.size * sizeof(uint32))
            debugNew(array95.id)
         let loopLimit31 = (50) div (5) + 1
         for loop31 in 0 ..< loopLimit31:
            discard
            for i in 0 ..< array94.size:
               dec array94.data[i]
            for i in 0 ..< array94.size:
               if array94.data[i] == 22:
                  return array94
         var params0: ArrayTParam
         params0.size = 2
         params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
         params0.data[0] = array94
         params0.data[1] = array95
         var array96: ptr ArrayT = func54(addr params0, getPath(), loopsFactor)
         debugReturn(array96.id)
         dealloc(params0.data)
         dec array96.refC
         if array96.refC == 0:
            dealloc(array96.data)
            dealloc(array96)
            debugFree(array96.id)
         dec array95.refC
         if array95.refC == 0:
            dealloc(array95.data)
            dealloc(array95)
            debugFree(array95.id)
      else:
         discard
         var params0: ArrayTParam
         params0.size = 1
         params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
         params0.data[0] = array94
         var array97: ptr ArrayT = func50(addr params0, getPath(), loopsFactor)
         debugReturn(array97.id)
         dealloc(params0.data)
         dec array97.refC
         if array97.refC == 0:
            dealloc(array97.data)
            dealloc(array97)
            debugFree(array97.id)
      var array98: ptr ArrayT
      if pCounter > 0:
         dec pCounter
         array98 = vars.data[pCounter]
         inc array98.refC
         debugCopy(array98.id)
      else:
         array98 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
         array98.size = 258
         array98.refC = 1
         array98.id = 98
         array98.data = cast[ptr UncheckedArray[uint32]](alloc(array98.size * sizeof(uint32)))
         zeroMem(array98.data, array98.size * sizeof(uint32))
         debugNew(array98.id)
      var params0: ArrayTParam
      params0.size = 2
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      params0.data[0] = array94
      params0.data[1] = array98
      var array99: ptr ArrayT = func51(addr params0, loopsFactor)
      debugReturn(array99.id)
      dealloc(params0.data)
      if (PATH0 and 2'u64) != 0:
         discard
         var array100: ptr ArrayT
         if pCounter > 0:
            dec pCounter
            array100 = vars.data[pCounter]
            inc array100.refC
            debugCopy(array100.id)
         else:
            array100 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
            array100.size = 62
            array100.refC = 1
            array100.id = 100
            array100.data = cast[ptr UncheckedArray[uint32]](alloc(array100.size * sizeof(uint32)))
            zeroMem(array100.data, array100.size * sizeof(uint32))
            debugNew(array100.id)
         let loopLimit32 = (50) div (5) + 1
         for loop32 in 0 ..< loopLimit32:
            discard
            for i in 0 ..< array94.size:
               dec array94.data[i]
            for i in 0 ..< array94.size:
               if array94.data[i] == 36:
                  return array94
         var params1: ArrayTParam
         params1.size = 4
         params1.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params1.size * sizeof(ptr ArrayT)))
         params1.data[0] = array94
         params1.data[1] = array98
         params1.data[2] = array99
         params1.data[3] = array100
         var array101: ptr ArrayT = func60(addr params1, getPath(), loopsFactor)
         debugReturn(array101.id)
         dealloc(params1.data)
         dec array101.refC
         if array101.refC == 0:
            dealloc(array101.data)
            dealloc(array101)
            debugFree(array101.id)
         dec array100.refC
         if array100.refC == 0:
            dealloc(array100.data)
            dealloc(array100)
            debugFree(array100.id)
      else:
         discard
         var params1: ArrayTParam
         params1.size = 3
         params1.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params1.size * sizeof(ptr ArrayT)))
         params1.data[0] = array94
         params1.data[1] = array98
         params1.data[2] = array99
         var array102: ptr ArrayT = func52(addr params1, loopsFactor)
         debugReturn(array102.id)
         dealloc(params1.data)
         dec array102.refC
         if array102.refC == 0:
            dealloc(array102.data)
            dealloc(array102)
            debugFree(array102.id)
      for i in 0 ..< array99.size:
         if array99.data[i] == 71:
            return array99
      dec array99.refC
      if array99.refC == 0:
         dealloc(array99.data)
         dealloc(array99)
         debugFree(array99.id)
      dec array98.refC
      if array98.refC == 0:
         dealloc(array98.data)
         dealloc(array98)
         debugFree(array98.id)
   for i in 0 ..< array94.size:
      dec array94.data[i]
   for i in 0 ..< array94.size:
      if array94.data[i] == 81:
         return array94
   return array94

